iina/PlaylistViewController.swift

```diff
@@ -143,6 +153,151 @@ class PlaylistViewController: NSViewController, NSTableViewDataSource {
       return nil
     }
   }
+  
+  // MARK: - Drag and Drop
+  
+  
+  func tableView(_ tableView: NSTableView, writeRowsWith rowIndexes: IndexSet, to pboard: NSPasteboard) -> Bool {
+    let data = NSKeyedArchiver.archivedData(withRootObject: [rowIndexes])
+    pboard.declareTypes([IINAPlaylistItemType], owner:self)
+    pboard.setData(data, forType:IINAPlaylistItemType)
+    return true
+  }
+
+
+  func tableView(_ tableView: NSTableView, validateDrop info: NSDraggingInfo, proposedRow row: Int, proposedDropOperation dropOperation: NSTableViewDropOperation) -> NSDragOperation {
+    let pasteboard = info.draggingPasteboard()
+
+    if let fileNames = pasteboard.propertyList(forType: NSFilenamesPboardType) as? [String] {
+      var hasItemToAdd: Bool = false
+      for path in fileNames {
+        let ext = (path as NSString).pathExtension
+        if !playerCore.supportedSubtitleFormat.contains(ext) {
+          hasItemToAdd = true
+          break
+        }
+      }
+
+      if !hasItemToAdd {
+        return []
+      }
+      playlistTableView.setDropRow(row, dropOperation: .above)
+      return .copy
+    }
+    if let _ = pasteboard.propertyList(forType: IINAPlaylistItemType) {
+      playlistTableView.setDropRow(row, dropOperation: .above)
+      return .move
+    }
+    return []
+}
+
+  func tableView(_ tableView: NSTableView, acceptDrop info: NSDraggingInfo, row: Int, dropOperation: NSTableViewDropOperation) -> Bool {
+    let pasteboard = info.draggingPasteboard()
+
+    if let rowData = pasteboard.data(forType: IINAPlaylistItemType) {
+      var dataArray = NSKeyedUnarchiver.unarchiveObject(with: rowData) as! Array<IndexSet>
+      let indexSet = dataArray[0]
+
+      let playlistCount = tableView.numberOfRows - 1
+      var order: Array<Int> = Array(0...playlistCount)
+      var finalRow = row
+
+      for selectedRow in indexSet.reversed() {
+        if selectedRow < row {
+          finalRow -= 1
+        }
+        order.remove(at: selectedRow)
+      }
+
+      for selectedRow in indexSet.reversed() {
+        order.insert(selectedRow, at: finalRow)
+      }
+
+      var fileList: [String] = []
+      var playing: Int = 0
```

Comment:
Sometimes it's better to clarify the type for readability, but `Int` is a simple type (that can be inferred easily by complier) and also the default type for integer literals, so better omit for simplicity.

```swift
var x = 0  // x is Int by default
var x: UInt32 = 0  // other int types
```